myhoard/controller.py

```diff
@@ -1310,67 +1409,143 @@ def _process_removed_binlogs(self, binlogs):
+        completed_backups = sort_completed_backups(
+            [backup for backup in self.state["backups"] if not backup["recovery_site"]]
+        )
+        if not completed_backups:
+            return Controller.NoPurgeReason.no_completed_backups
+
+        # For simplicity only ever drop one backup here. This function
+        # is called repeatedly so if there are for any reason more backups
+        # to drop they will be dropped soon enough
+        backup = completed_backups[0]
+
+        def is_backup_incremental(b: Backup) -> bool:
+            return b["basebackup_info"].get("incremental", False)
 
-        if non_broken_backups_count <= self.backup_settings["backup_count_max"] < len(purgeable):
+        # If the first backup is incremental, it's useless without a preceding full backup, so drop it
+        if is_backup_incremental(backup):
+            return backup
+
+        # Do not consider broken backups for the count, they will still be purged,
+        # but we should only purge when the count of non-broken backups has exceeded the limit.
+        non_broken_backups_count = Controller.count_non_broken_backups(completed_backups)
+        if non_broken_backups_count <= self.backup_settings["backup_count_max"] < len(completed_backups):
+            broken_backups_count = len(completed_backups) - non_broken_backups_count
             self.log.info(
-                "Backup count %s is above max allowed, but %s are broken, not dropping",
-                len(purgeable),
+                "Backup count %s is above max allowed, but excluding broken amount %s, still not more than max,"
+                " not dropping",
+                len(completed_backups),
                 broken_backups_count,
             )
-            return
+            return Controller.NoPurgeReason.within_max_amount
 
-        if non_broken_backups_count <= self.backup_settings["backup_count_min"]:
-            return
+        # Create a list of incremental and dependant backups, they are interconnected and needed for later checks
+        incremental_backups = []
+        for b in completed_backups[1:]:
+            if not is_backup_incremental(b):
+                break
+            incremental_backups.append(b)
 
-        # For simplicity only ever drop one backup here. This function
-        # is called repeatedly so if there are for any reason more backups
-        # to drop they will be dropped soon enough
-        purgeable = sort_completed_backups(purgeable)
-        backup = purgeable[0]
+        dependant_backups = [backup] + incremental_backups
+        # We should always keep min amount of backups, check if that works if we drop this and potentially following
+        # incremental backups
+        if (
+            non_broken_backups_count - Controller.count_non_broken_backups(dependant_backups)
+            < self.backup_settings["backup_count_min"]
+        ):
+            return Controller.NoPurgeReason.min_amount_required
```

Comment:
Same here, I think `backup_count_min` should mean count of full backups.

Reply:
Same answer I guess :)